stubs/toml/toml/decoder.pyi

```diff
@@ -0,0 +1,79 @@
+import sys
+from _typeshed import SupportsRead
+from typing import Any, Callable, Generic, MutableMapping, Pattern, Protocol, Text, TypeVar, overload
+
+_MutableMappingT = TypeVar("_MutableMappingT", bound=MutableMapping[str, Any])
+_KT_contra = TypeVar("_KT_contra", contravariant=True)
+_VT = TypeVar("_VT")
+
+if sys.version_info >= (3, 0):
+    from pathlib import PurePath
+
+    FNFError = FileNotFoundError
+    _PathLike = str | bytes | PurePath
+else:
+    FNFError = IOError
+    _PathLike = str | bytes
```

Comment:
This doesn't do what you intended it to do. On Python 2, `str` and `bytes` are the same thing, and `unicode` is different. Maybe you meant `Text`, which accepts both `str` and `unicode` (if I remember correctly)?

Reply:
I got confused because there's no much documentation about `basestring` but since `unicode` and `str` inherent from it I think you are right

here's the relevant part in `toml`'s source code if there's a best practice to deal with `basestring`

```python
def _ispath(p):
    if isinstance(p, (bytes, basestring)):
        return True
    return _detect_pathlib_path(p)  # check for pathlib.PurePath

def load(f, _dict=dict, decoder=None):
    if _ispath(f):
        ...  # open `f` and read it using io.open
    elif isinstance(f, list):
        ...
    else:
        return loads(f.read(), _dict, decoder)
```

Reply:
```suggestion
    _PathLike = Text
```